pkg/sessiontxn/staleread/util.go

```diff
@@ -40,7 +41,12 @@ func CalculateAsOfTsExpr(ctx context.Context, sctx sessionctx.Context, tsExpr as
 		// this can be more accurate than `time.Now() - staleness`, because TiDB's local time can drift.
 		return sctx.GetStore().GetOracle().GetStaleTimestamp(ctx, oracle.GlobalTxnScope, 0)
 	})
-	tsVal, err := expression.EvalAstExpr(sctx, tsExpr)
+	expr, err := expression.RewriteAstExpr(sctx, tsExpr, nil, nil, false)
```

Comment:
Now `EvalAstExpr` does not support `ParamMarker`, we use `RewriteAstExpr` instead